core/src/main/java/org/arquillian/smart/testing/impl/TestExecutionPlannerLoaderImpl.java

```diff
@@ -21,14 +22,23 @@
         this.projectDir = projectDir;
     }
 
-    public TestExecutionPlanner getPlannerForStrategy(String strategy) {
+    public TestExecutionPlanner getPlannerForStrategy(String strategy, boolean autocorrect) {
 
         if (availableStrategies.isEmpty()) {
             loadStrategies();
         }
 
         if (availableStrategies.containsKey(strategy)) {
             return availableStrategies.get(strategy).create(projectDir, verifier);
+        } else {
+            if (autocorrect) {
+                final StringSimilarityCalculator stringSimilarityCalculator = new StringSimilarityCalculator();
+                final String closestMatch =
+                    stringSimilarityCalculator.findClosestMatch(strategy, availableStrategies.keySet());
+                if (availableStrategies.containsKey(closestMatch)) {
```

Comment:
is this if necessary?

Reply:
btw, is there any benefit of creating the instance of `StringSimilarityCalculator` using the constructor and not using it as static class/method?

Reply:
Well it was initially implemented as this

Reply:
About condition, well I think it is a safe condition no avoid any NPE